### PR TITLE
media-sound/jack: version bump to 4.0.0 pre-release

### PR DESCRIPTION
**Pull request by madmartin**
initial python3 version, but still WIP
basic functionality is implemented

Closes: https://bugs.gentoo.org/734572
Package-Manager: Portage-2.3.103, Repoman-2.3.23
Signed-off-by: Martin Dummer <martin.dummer@gmx.net>

### CODE REVIEW

**Comment by thesamesam** [comment]
Hi, thanks for working on this. If it's still a WIP, mark it as a draft, but we will still give feedback to you.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:6]: Our policy right now is "as many Pythons (3.x) as possible", so 3.6, 3.7, 3.8, 3.9 if it works/tests pass with all, please!

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:20]: You can drop this.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:10]: Not actually tested but we may need DISTUTILS_USE_SETUPTOOLS=rdepend because of [this](https://github.com/jack-cli-cd-ripper/jack/blob/python3-mb/setup.py#L27) (see [here](https://dev.gentoo.org/~mgorny/python-guide/distutils.html#different-build-system-variations)).

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:17]: Might be GPL-2+ based on the README and header of file(s?)

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:15]: Replace ${PN}-${PV} with ${P} :)

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:6]: Hello @thesamesam , thanks for your comments, you're the fastest reviewer I have seen as proxy-maintainer.

Let me say that I know some of your comments, I want mainly indicate that I am working on this ebuild.
I will test as many python versions as I can, and update the ebuild accordingly.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:20]: What exactly you're talking about? any variables assigned to "" (empty)?

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:17]: Will investigate...

Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:17]: yes, GPL-2+

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:11]: done

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (madmartin) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild]: I tested python 3.6 3.7 3.8 and found them working.

**Comment by madmartin** [comment]
Hello @thesamesam , I revised the ebuild after thoroughly testing the software. It works quite good, IMHO it is okay to put it into the tree.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (mgorny) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:31]: These all seem like RDEPEND. Are they really used at build time?

Don't leave empty *DEPEND around.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (mgorny) [inline media-sound/jack/jack-4.0.0_pre20200706.ebuild:33]: Move this immediately below SRC_URI, as the two usually go together.

**Review by mgorny — CHANGES_REQUESTED** [review]

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild]: Have a look [here](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-610007.2), yeah, IUSE can go.

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild:11]: Thanks!

**Comment by madmartin** [comment]
OMG someone kicked jack-3.1.1-r3.ebuild out in the meantime. So I must re-add metadata.xml an rebase....

**Comment by mgorny** [comment]
> OMG someone kicked jack-3.1.1-r3.ebuild out in the meantime. So I must re-add metadata.xml an rebase....

I'm sorry about that. I thought updating the mask to say `<4_pre` would be sufficient for people to figure out progress is being made but apparently it wasn't.

**Comment by madmartin** [comment]
> > OMG someone kicked jack-3.1.1-r3.ebuild out in the meantime. So I must re-add metadata.xml an rebase....
> 
> I'm sorry about that. I thought updating the mask to say `<4_pre` would be sufficient for people to figure out progress is being made but apparently it wasn't.

@mgorny @thesamesam  No Problem. I updated the PR as requested, please review again. Maybe the "do not merge" label can be removed....

File: media-sound/jack/jack-4.0.0_pre20200706.ebuild
Comment (thesamesam) [reply media-sound/jack/jack-4.0.0_pre20200706.ebuild]: Fab.

**Comment by thesamesam** [comment]
Not forgot about you, been re-doing testing environment. Thank you!

**Pull request closed without merging**
